lib/srv/ctx.go

```diff
@@ -676,22 +678,82 @@ func (c *ServerContext) String() string {
 	return fmt.Sprintf("ServerContext(%v->%v, user=%v, id=%v)", c.ServerConn.RemoteAddr(), c.ServerConn.LocalAddr(), c.ServerConn.User(), c.id)
 }
 
-// ExecCommand takes a *ServerContext and extracts the parts needed to create
-// an *execCommand which can be re-sent to Teleport.
-func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
-	var pamEnabled bool
-	var pamServiceName string
-	var pamUseAuth bool
+func (c *ServerContext) getPAMConfig() (*PAMConfig, error) {
+	if c.srv.Component() != teleport.ComponentNode {
+		return &PAMConfig{Enabled: false}, nil
+	}
+
+	accessPoint := c.srv.GetAccessPoint()
+	clusterPAMConfig, err := accessPoint.GetPAMConfig(context.TODO())
```

Comment:
I'd consider using `c.cancelContext` to be able to abort this call if the server context is cancelled.